src/App.js

```diff
@@ -21,95 +21,41 @@ const messages = {
 addLocaleData([...en, ...es]);
 
 class App extends Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
+
+    // TODO: Consider continue determining here default instance or moving all upside,
+    // forcing inject dependencies always
+    this.sessionManager =
+      (props.dependencies && props.dependencies.sessionManager) ||
+      new OnlineSessionManager(
+        new HttpDopplerMvcClient(axios, process.env.REACT_APP_API_URL),
+        60000,
+      );
+
     //TODO: this hardcoded data will depend by the app language
     const locale = navigator.language.toLowerCase().split(/[_-]+/)[0] || 'en';
     this.state = {
-      user: null,
-      loginSession: {},
+      dopplerSession: this.sessionManager.session,
       i18n: {
         locale: locale,
         messages: flattenMessages(messages[locale]),
       },
     };
   }
 
-  componentWillMount() {
-    this.getUserData();
-    this.interval = setInterval(() => {
-      this.getUserData();
-    }, 60000);
+  componentDidMount() {
```

Comment:
why change willMount to didMount?? no need initialize session manager first?

Reply:
As I understand, the usage of willMount is discouraged. What do you think?

Reply:
Pregunta no relacionada, ¿por que son los dobles signos de interrogación?

> ```
> why change willMount to didMount??
>                                 ^^
> ```

Creo que querés poner una especie de enfasis pero no se a que.

Reply:
se me fue el otro signo de pregunta jaja, iba uno solo. igual creo al estar el `loading..` no hay problema de usar el otro

Reply:
Ok, antes de seguir por esto voy a investigar para entender bien como funciona.

Idealmente, creo que debería estar en el constructor ya que cuanto antes inicie la llamada Ajax, mejor. Pero tengo entendido que no es buena idea, "por algo de react". Por itro lado también leí que estaba desaconsejado el uso de _componentWillMount_ pero tampoco entendí bien la razón. 

Antes de seguir lo investigo. Si vos tenés más claro como funciona, pros y cons, avisame y me explicás.

Reply:
https://daveceddia.com/where-fetch-data-componentwillmount-vs-componentdidmount/

Es verdad habria que usar el component didmount no el constructor